Fix incorrect FCE counter

ethanliu24 opened the pull request:
## Proposed Changes

_(Describe your changes here. Also describe the motivation for your changes: what problem do they solve, or how do they improve the application or codebase? If this pull request fixes an open issue, [use a keyword to link this pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)_

Currently when a course node is clicked, it adds 0.5 credits to the FCE counter. This PR differentiates between year and half year courses such that when a year course is clicked, 1.0 credit is added and when half year courses are clicked, 0.5 credits are added.
...

<details>
<summary>Screenshots of your changes (if applicable)</summary>

</details>

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  | X        |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |          |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [X] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [X] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.
- [ ] I have updated the project documentation, if applicable.
  - This is **required** for new features.
- [ ] If this is my first contribution, I have added myself to the list of contributors.
- [x] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

- [x] I have verified that the CircleCI checks have passed.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

david-yz-liu reviewed (commented):
@ethanliu24 the code change is great. Please add some additional test cases for deselecting the course as well (to decrement the credit count). This is the other branch of the code that you changed.

ethanliu24 commented:
I have added more test cases. I noticed that `componentDidMount` also involves counting credits and when page is refreshed, it counts it incorrectly. I have fixed that and added a test for it. Also, counting FCE by checking for letter "y" or "h" won't work on the Statistics graph, since there's no y nor h in the node name. I'm assuming that page is only a demo for now, but would that be fine with you?

david-yz-liu reviewed (approved):
@ethanliu24 great work! The way you've implemented the helper function, which treats a "missing" `'h'` or `'y'` as a 0.5 credit course, is appropriate. It's fine that this will be the behaviour for all stats courses, and it's great that you noticed this.
